core/src/main/scala/org/apache/spark/SecurityManager.scala

```diff
@@ -529,7 +529,8 @@ private[spark] class SecurityManager(
       return
     }
 
-    if (sparkConf.get(SparkLauncher.SPARK_MASTER, null) != "yarn") {
+    val master = sparkConf.get(SparkLauncher.SPARK_MASTER, "")
```

Comment:
I'm not sure this will work for local-cluster. Try `local-cluster[2, 1, 1024]` as the master and see what happens.

Reply:
Nice catch! I've enhanced the test and fixed the issue.